documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -351,4 +353,258 @@ Control characters are not allowed in a specification.
+
+### <a name="R1001" />R1001 OperationIdNounVerb
+**Output Message**: Per the Noun_Verb convention for Operation Ids, the noun '{0}' should not appear after the underscore.
+
+**Description**: OperationId should be of the form `Noun_Verb`. 
+
+**Why the rule is important**: AutoRest breaks the operation id into its `Noun` and `Verb` where `Noun` becomes name of the operations class and the `Verb` becomes the name of the method in that class. Not adhering to this format can either cause AutoRest to fail or can generate semantically incorrect SDK.
+
+**How to fix the violation**: Ensure operationId is of the form `Noun_Verb`.
+
+**Bad Example**:
+```
+Activate_Certificate
+CertificateActivate
+```
+**Good Example**:
+```
+Certificate_Activate
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2055" />R2055 OneUnderscoreInOperationId
+**Output Message**: Only 1 underscore is permitted in the operation id, following Noun_Verb conventions.
+
+**Description**: An operationId can have exaclty one underscore, not adhering to it can cause errors in code generation.
+
+**Why the rule is important**: Given an operationId of the form `Noun_Verb`, AutoRest breaks the operation id into its `Noun` and `Verb` where `Noun` becomes name of the operations class and the `Verb` becomes the name of the method in that class. Not adhering to this format can cause AutoRest to fail during code generation.
+
+**How to fix the violation**: Ensure operationId is of the form `Noun_Verb` and contains exactly one underscore.
+
+**Bad Example**:
+```
+Activate_Primary_Certificate
+```
+**Good Example**:
+```
+PrimaryCertificate_Activate
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R3023" />R3023 OperationsAPIImplementation
+**Output Message**: Operations API must be implemented for '{0}'.
+
+**Description**: Per ARM guidelines, each RP must expose an operations API that returns information about all the operations available with the service.
+
+**Why the rule is important**: For better user experience. Users can query the service to get a list of all possible operations on a service and decide what they have to do.
+
+**How to fix the violation**: Add an operations API endpoint (if not already present) and add details regarding this endpoint in the corresponding OpenAPI document. Examples can be found for most RPs in this repo.
```

Comment:
could we include an example of this? The actual swagger for this operation is mostly the same for folks, so an example would help with copy/paste.